lib/bolt/node/ssh.rb

```diff
@@ -171,11 +171,19 @@ def write_remote_file(source, destination)
     end
 
     def make_tempdir
-      result = execute('mktemp -d')
+      tmppath = nil
+      if @tmpdir
+        tmppath = "#{@tmpdir}/#{(('a'..'z').to_a + ('A'..'Z').to_a).shuffle.first(12).join}"
```

Comment:
Is this secure, maybe `SecureRandom.uuid` instead?

Reply:
no I was trying to decide if it mattered. I guess you could prevent the expected script from actually being run.

Reply:
ah, I'm more concerned about TOCTOU race conditions if the `tmppath` is writable by less privileged users.